lib/OpenQA/CacheService/Client.pm

```diff
@@ -31,18 +32,19 @@ has ua => sub ($self) {
     # note: The socket_options function has only been added in Mojolicious 8.72. For older versions we still rely
     #       on the monkey_patch within the BEGIN block of Worker.pm. It could be removed when we stop supporting older
     #       Mojolicious versions.
+    return $ua unless _is_ipv4_addr($self->actual_host);
     my %cache_service_address = (
         family => AF_INET,
         protocol => IPPROTO_TCP,
         socktype => SOCK_STREAM,
-        addr => pack_sockaddr_in(service_port('cache_service'), inet_aton('127.0.0.1')));
+        addr => pack_sockaddr_in(service_port('cache_service'), inet_aton($self->actual_host)));
     $ua->socket_options->{PeerAddrInfo} = [\%cache_service_address] if $ua->can('socket_options');
     return $ua;
 };
 
 sub set_port ($self, $port) {
-    $self->ua->socket_options->{PeerAddrInfo}->[0]->{addr} = pack_sockaddr_in($port, inet_aton('127.0.0.1'))
-      if $self->ua->can('socket_options');
+    $self->ua->socket_options->{PeerAddrInfo}->[0]->{addr} = pack_sockaddr_in($port, inet_aton($self->actual_host))
+      if $self->ua->can('socket_options') && _is_ipv4_addr($self->actual_host);
```

Comment:
The `if $self->ua->can('socket_options')` check is obsolete now since we depend on Mojolicious 9.20. You could just remove that.